extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/iam/IamServiceImpl.java

```diff
@@ -196,4 +222,25 @@ private Supplier<IamCredentialsClient> defaultIamCredentialsClientSupplier() {
             };
         }
     }
+
+    static class ApplicationDefaultCredentials implements AccessTokenProvider {
```

Comment:
this should be private. And it can be a `record`

Reply:
Done